fix(bb.js): pass console.log logger in tests to avoid Jest noise

johnathan79717 opened the pull request:
Pass `console.log` as custom logger in bb.js tests instead of relying on the default `console.error` logger.

Jest adds stack traces to `console.error` output, making test logs verbose and hard to read. Using `console.log` avoids this while keeping `console.error` as the default for production (to avoid stdout pipeline interference).

Also adds `afterAll` cleanup to poseidon benchmark test to properly destroy backends and prevent "Jest did not exit" warnings.

johnathan79717 committed:
chore(bb.js): pass console.log logger in tests to avoid Jest noise

Pass console.log as the logger in bb.js tests to avoid Jest adding
stack traces to console.error output. The default logger stays on
stderr (console.error) to avoid interfering with stdout pipelines.

ludamad commented on aztec-up/test/amm_flow.sh:
```diff
@@ -35,6 +35,7 @@ while ! curl -fs localhost:8080/status &>/dev/null; do sleep 1; done
 
 canonical_sponsored_fpc_address=$(aztec \
   get-canonical-sponsored-fpc-address \
```
This is a breaking change - we can't put stuff in stdout if we have an API that is meant to be captured as-is. We'd have to make sure that bbjs logger is consistently overridden

johnathan79717 replied:
Good point. Doing it the other way around now, passing console.log to jest tests to avoid it treating every log as errors.

johnathan79717 committed:
revert: restore console.error in default logger

johnathan79717 commented:
Didn't realize there was still test failures before re-requesting review.

johnathan79717 committed:
fix(bb.js): add afterAll cleanup and logger to poseidon benchmark test

johnathan79717 committed:
fix(bb.js): close readline interfaces on destroy to prevent post-test logging

Close stdout/stderr readline interfaces before killing the bb process.
This prevents "Cannot log after tests are done" warnings when bb logs
its shutdown message after Jest has finished.
